components/style/build_gecko.rs

```diff
@@ -517,6 +517,49 @@ mod bindings {
         write_binding_file(builder, structs_file(build_type), &fixups);
     }
 
+    pub fn setup_logging() {
+        use log;
+        use std::fs;
+
+        struct BuildLogger {
+            file: Option<Mutex<fs::File>>,
+            filter: String,
+        }
+
+        impl log::Log for BuildLogger {
+            fn enabled(&self, meta: &log::LogMetadata) -> bool {
+                self.file.is_some() && meta.target().contains(&self.filter)
+            }
+
+            fn log(&self, record: &log::LogRecord) {
+                if !self.enabled(record.metadata()) {
+                    return;
+                }
+
+                let mut file = self.file.as_ref().unwrap().lock().unwrap();
+                let _ =
```

Comment:
I have no real preference, I thought that we shouldn't panic just for failing to log to disk, but...